src/scores/continuous/consistent_impl.py

```diff
@@ -0,0 +1,262 @@
+        phi_prime: a subderivative of `phi`, accepting a single array like argument.
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the consistent expectile score. All other dimensions will be preserved. As a
+            special case, 'all' will allow all dimensions to be reduced. Only one
+            of `reduce_dims` and `preserve_dims` can be supplied. The default behaviour
+            if neither are supplied is to reduce all dims.
+        preserve_dims: Optionally specify which dimensions to preserve when calculating
+            the consistent quantile score. All other dimensions will be reduced. As a special case, 'all'
+            will allow all dimensions to be preserved. In this case, the result will be in
+            the same shape/dimensionality as the forecast, and the errors will be the consistent quantile
+            score at each point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely. Only one of `reduce_dims`
+            and `preserve_dims` can be supplied. The default behaviour if neither are supplied
+            is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+
+    Returns:
+        array of (mean) scores that is consistent for alpha-expectile functional,
+        with the dimensions specified by `dims`. If `dims` is `None`, the returned DataArray will have
+        only one entry, the overall mean score.
+
+    Raises:
+        ValueError: if `alpha` is not strictly between 0 and 1.
+
+    Note:
+        .. math::
+
+            S(x, y) =
+            \\begin{cases}
+            (1 - \\alpha)(\\phi(y) - \\phi(x) - \\phi'(x)(y-x)), & y < x \\\\
+            \\alpha(\\phi(y) - \\phi(x) - \\phi'(x)(y-x)), & x \\leq y
+            \\end{cases}
+
+        where
+
+            - :math:`x` is the forecast
+            - :math:`y` is the observation
+            - :math:`\\alpha` is the expectile level
+            - :math:`\\phi` is a convex function of a single variable
+            - :math:`\\phi'` is the subderivative of :math:`\\phi`
+            - :math:`S(x,y)` is the score.
+
+        Note that if :math:`\\phi` is differentiable then `\\phi'` is its derivative.
+
+    References:
+        -   Gneiting, T. (2011). Making and Evaluating Point Forecasts. Journal of the 
+            American Statistical Association, 106(494), 746–762. 
+            https://doi.org/10.1198/jasa.2011.r10138
+        -   Taggart, R. (2021). Evaluation of point forecasts for extreme events using 
+            consistent scoring functions. Quarterly Journal of the Royal Meteorological 
+            Society, 148(742), 306–320. https://doi.org/10.1002/qj.4206
+
+    """
+    check_alpha(alpha)
+
+    if preserve_dims or reduce_dims:
```

Comment:
It should be safe to just go ahead and call gather_dimensions regardless, but perhaps this is faster. The upside of calling it regardless is that if gather_dimensions ever includes additional checks (e.g. based on just fast and obs), then those won't be skipped. Perhaps it's an academic concern only at this point.